VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/TryCastWithoutUsingAsNotNull/TryCastWithoutUsingAsNotNullAnalyzer.cs

```diff
@@ -99,7 +99,10 @@ private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
                 }
             };
 
-            foreach (var expression in asExpressions.Concat<ExpressionSyntax>(castExpressions))
+            var expressions = new List<ExpressionSyntax>(asExpressions);
```

Comment:
Initialize the list with `new List<ExpressionSyntax(asExpressions.Length + castExpressions.Length)` and then use `AddRange` twice. This will prevent a resizing of the internal list because it already had the correct size from the start